src/main/java/practice/StreamPractice.java

```diff
@@ -50,7 +70,14 @@ public List<Person> selectMenByAge(List<Person> peopleList, int fromAge, int toA
      */
     public List<Person> getWorkablePeople(int fromAge, int femaleToAge,
                                           int maleToAge, List<Person> peopleList) {
-        return Collections.emptyList();
+        return peopleList.stream()
+                .filter(person -> ((person.getSex() == Person.Sex.MAN
+                        && person.getAge() > fromAge
+                        && person.getAge() <= maleToAge)
+                        || (person.getSex().equals(Person.Sex.WOMAN)
+                        && person.getAge() >= fromAge
+                        && person.getAge() <= femaleToAge)))
```

Comment:
```suggestion
Predicate<Person> malePredicate = p -> p.getSex() == Person.Sex.MAN
                && p.getAge() >= fromAge && p.getAge() <= maleToAge;
Predicate<Person> femalePredicate = p -> p.getSex() == Person.Sex.WOMAN
                && p.getAge() >= fromAge && p.getAge() <= femaleToAge;

return peopleList.stream()        
                             .filter(malePredicate.or(femalePredicate))
```
You can do it in this way